addons/im_livechat/controllers/main.py

```diff
@@ -97,34 +97,32 @@ def get_session(self, channel_id, previous_operator_id=None, chatbot_script_id=N
             country = request.env["res.country"].search(
                 [("code", "=", request.geoip.country_code)], limit=1
             )
-        if chatbot_script_id and chatbot_script_id in livechat_channel.rule_ids.chatbot_script_id.ids:
-            chatbot_script = (
-                request.env["chatbot.script"]
-                .sudo()
-                .with_context(lang=request.env["chatbot.script"]._get_chatbot_language())
-                .search([("id", "=", chatbot_script_id)])
-            )
-        else:
-            agent = livechat_channel._get_operator(
-                previous_operator_id=previous_operator_id,
-                lang=request.cookies.get("frontend_lang"),
-                country_id=country.id,
-            )
-        if not chatbot_script and not agent:
+        operator_lookup_params = {
+            'previous_operator_id': previous_operator_id,
+            'chatbot_script_id': chatbot_script_id,
+            'country_id': country.id,
+            'lang': request.cookies.get("frontend_lang"),
+            **extra_operator_lookup_params,
+        }
+        operator_info = livechat_channel._get_operator_info(**operator_lookup_params)
```

Comment:
we don't need a separate `operator_lookup_params` to spread it, just pass the named parameters as before and add **extra_operator_lookup_params to the function